### PR TITLE
Fixed issue of wrong prefixes under multiplication

### LINKED ISSUE #13033: Prefixes wrong under multiplication
```python
from sympy.physics.units import *
3*kilo  # works
kilo*3  # raises an exception
```

### PR DESCRIPTION
**Pull request by Subhash-Saurabh**
Fixes  #13033
 **Before**
```
>>> from sympy.physics.units import *
>>> 3*kilo
3*Prefix(kilo, k, 3, 10)
>>> kilo*3
AttributeError: 'int' object has no attribute 'scale_factor'
```
**Now**
```
>>> from sympy.physics.units import *
>>> 3*kilo
3*Prefix(kilo, k, 3, 10)
>>> kilo*3
3*Prefix(kilo, k, 3, 10)
```

### CODE REVIEW

**Comment by Subhash-Saurabh** [comment]
@Upabjojr can you please take a look at this PR.

File: sympy/physics/units/tests/test_prefixes.py [hunk 1]
```diff
@@ -25,6 +25,8 @@ def test_prefix_operations():
     m = Quantity("meter", 1, 6)
     assert dodeca / m == 12 / m
 
+    assert kilo*3 == kilo*3
```
Comment (Upabjojr) [inline sympy/physics/units/tests/test_prefixes.py]: This looks like a tautology.

File: sympy/physics/units/prefixes.py [hunk 2]
```diff
@@ -73,6 +73,9 @@ def __str__(self):
     __repr__ = __str__
 
     def __mul__(self, other):
+        if isinstance(sympify(other), Number):
+            return other*self
```
Comment (Upabjojr) [inline sympy/physics/units/prefixes.py]: This will ignore Python numbers (`Number` is a SymPy class).

Comment (Upabjojr) [reply sympy/physics/units/prefixes.py]: I would suggest something like 
```
if not isinstance(other, Prefix):
    super(Prefix, self).__mul__(other)
```

File: sympy/physics/units/tests/test_prefixes.py (hunk 1 above)
Comment (Upabjojr) [reply sympy/physics/units/tests/test_prefixes.py]: maybe
```
assert isinstance(kilo*3, Mul)
assert (kilo*3).args == (3, kilo)
```

File: sympy/physics/units/prefixes.py (hunk 2 above)
Comment (Subhash-Saurabh) [reply sympy/physics/units/prefixes.py]: After using `sympify(other)` all python numbers will get converted to sympy Numbers isn't it.
If I am missing something please point out.

Comment (Subhash-Saurabh) [reply sympy/physics/units/prefixes.py]: However making the changes suggested by you will make it work with symbols and other types as well.

**Comment by ylemkimon** [comment]
`__div__` seems to have the same problem.

**Comment by Upabjojr** [comment]
Eventually we will need to find a way not to overload `__mul__` in subtypes.

**Pull request merged**